plugin-maven/src/main/java/com/diffplug/spotless/maven/generic/Format.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.maven.generic;
+
+import java.util.Collections;
+import java.util.Set;
+
+import com.diffplug.spotless.maven.FormatterFactory;
+
+public class Format extends FormatterFactory {
+
+	@Override
+	public Set<String> defaultIncludes() {
+		return Collections.emptySet();
+	}
+
+	@Override
+	public String licenseHeaderDelimiter() {
+		// do not specify a default delimiter
+		return null;
+	}
+
+	public void addEndWithNewline(EndWithNewline endWithNewline) {
+		addStepFactory(endWithNewline);
+	}
+
+	public void addIndent(Indent indent) {
+		addStepFactory(indent);
+	}
+
+	public void addTrimTrailingWhitespace(TrimTrailingWhitespace trimTrailingWhitespace) {
+		addStepFactory(trimTrailingWhitespace);
+	}
+
+	public void addReplace(Replace replace) {
+		addStepFactory(replace);
+	}
+
+	public void addReplaceRegex(ReplaceRegex replaceRegex) {
+		addStepFactory(replaceRegex);
+	}
+}
```

Comment:
So, `FormatterFactory.java` has exactly one subclass, `Format.java`, and every other format (e.g. `Java`, `Scala`) will extend `Format.java` and not `FormatterFactory.java`, is that correct?  I wonder if `Format` and `FormatterFactory` ought to be merged?  Thoughts @lutovich?

Reply:
Extending `Format` was quite a last-minute change. Haven't thought about merging those classes, but I agree that this should be considered. Which class should I delete (merge into the other)?

Reply:
Keep them separate for now.  Hopefully @lutovich will have a chance to take a look.

Reply:
Am I right that `Format` is needed to keep all injectable generic steps in one class, which is reused for `Java`, `Scala` and `AbstractSpotlessMojo`?

Reply:
Would it make sense to keep `Format` and `FormatterFactory` separate but move all `#add*` methods to `FormatterFactory`? Then `Java` and `Scala` can keep extending `FormatterFactory`.

Reply:
The class `Format`  aggregates all generic steps, which can be used to build custom formats, but also for `<java>`, `<scala>`, `<kotlin>` or `<markdown>` if you like. If using a formatter step like eclipse or googleFormat this won't make any sense most of the time. But formatting an markdown file might use some of the generic steps.

Moreover IMHO the maven-plugin should support the same feature set as the gradle-plugin does. And the gradle-plugin supports it this way.

So one could setup the formatting like this
```xml
<configuration>
<!-- global defaults -->
  <encoding>UTF-8</encoding>

  <!-- custom format for properties -->
  <format>
    <inlcudes>
      <include>src/**/resources/**/*.properties</include>
    </includes>
    <endWithNewline />
    <encoding>ISO 8859-1</encoding>
  </format>

  <!-- custom format for java files -->
  <java>
    <endWithNewline />
    <trimTrailingWhitespace />
  </java>
</configuration>
```